engine/src/flutter/BUILD.gn

```diff
@@ -178,11 +178,11 @@ group("unittests") {
     ]
   }
 
-  # Compile all unittests targets if enabled.
+  if (enable_ios_unittests) {
+    public_deps += [ "//flutter/shell/platform/darwin/ios:ios_test_flutter" ]
+  }
+
   if (enable_unittests) {
-    if (is_ios) {
-      public_deps += [ "//flutter/shell/platform/darwin/ios:ios_test_flutter" ]
-    }
```

Comment:
This block was moved inside the `enable_unittests` conditional here:
https://github.com/flutter/flutter/pull/167530/files#diff-0009a2dc7028a9776b47e2251ca6b7154d4cd374466a074996d29cf881ab940aL181

See `testing.gni` below and note it's defined as:
```python
enable_unittests = current_toolchain == host_toolchain || is_fuchsia || is_mac
```

I think a reasonable followup might be to rename `enable_unittests` to something like `enable_portable_unittests` or similar -- would love any suggestions on a better name. :)

Reply:
This is... so confusing. <s>may be add a comment here or rename it? </s> nvm i saw your another comment.